### ISSUE TITLE
Sensors have incorrect hard coded z offsets

### ISSUE THREAD

**Issue by jalderman9** [NREL/bifacial_radiance#185]
Currently the virtual sensor locations have hard coded z offsets added to the calculated `zstart` value, which results in their positions changing relative to the module as the scene tilt angle changes:

```
        frontscan = {'xstart': xstart+xinc, 'ystart': ystart+yinc,
                     'zstart': zstart + zinc + 0.06,
                     'xinc':xinc, 'yinc': yinc,
                     'zinc':zinc , 'Nx': 1, 'Ny':sensorsy, 'Nz':1, 'orient':front_orient }
        backscan = {'xstart':xstart+xinc, 'ystart':  ystart+yinc,
                     'zstart': zstart + zinc - 0.03,
                     'xinc':xinc, 'yinc': yinc,
                     'zinc':zinc, 'Nx': 1, 'Ny':sensorsy, 'Nz':1, 'orient':back_orient }
```

I added some small spheres at the sensor coordinates to visualize this.  You can see that at high tilt angles the front sensors are too high up on the module (z offset too large):
![new-colorfront](https://user-images.githubusercontent.com/31673048/67413714-894d4780-f576-11e9-8c32-1f0d73a4e3fa.png)

At noon they are pretty close to correct, although unnecessarily high above the module surface:
![new-noon](https://user-images.githubusercontent.com/31673048/67413729-90745580-f576-11e9-84de-c36f291c2d9f.png)

The rear sensors are 30mm below the module rear surface in z, which is incorrect and could also result in one of them ending up inside a torque tube:
![new-colorback](https://user-images.githubusercontent.com/31673048/67413829-c285b780-f576-11e9-9b2d-19e0bea8ceb7.png)

To address this I fixed the offsets with separate `xstart`, `ystart`, and `zstart` values for the front and back sensors and took the module z dimension (which is hard coded to 20mm elsewhere) into account:

```
        # Axis of rotation Offset (if offset is not 0) for the front of the module
        x3 = (offset + modulez + 0.003) * np.sin(tilt*dtor) * np.sin((azimuth)*dtor)
        y3 = (offset + modulez) * np.sin(tilt*dtor) * np.cos((azimuth)*dtor)
        z3 = (offset + modulez + 0.003) * np.cos(tilt*dtor)
        
        # Axis of rotation offset for the back of the module
        x4 = (offset - 0.003) * np.sin(tilt*dtor) * np.sin((azimuth)*dtor)
        y4 = offset * np.sin(tilt*dtor) * np.cos((azimuth)*dtor)
        z4 = (offset - 0.003) * np.cos(tilt*dtor)

        xstartfront = x1 + x2 + x3 + originx
        xstartback = x1 + x2 + x4 + originx
        ystartfront = y1 + y2 + y3 + originy
        ystartback = y1 + y2 + y4 + originy
        zstartfront = height + z1 + z2 + z3
        zstartback = height + z1 + z2 + z4

        frontscan = {'xstart': xstartfront+xinc, 'ystart': ystartfront+yinc,
                     'zstart': zstartfront + zinc,
                     'xinc':xinc, 'yinc': yinc,
                     'zinc':zinc , 'Nx': 1, 'Ny':sensorsy, 'Nz':1, 'orient':front_orient }
        backscan = {'xstart': xstartback+xinc, 'ystart':  ystartback+yinc,
                     'zstart': zstartback + zinc,
                     'xinc':xinc, 'yinc': yinc,
                     'zinc':zinc, 'Nx': 1, 'Ny':sensorsy, 'Nz':1, 'orient':back_orient }
```

Now my sensors are closer to the module surface and in the correct locations.
New front sensor locations:
![new-colorfront](https://user-images.githubusercontent.com/31673048/67415704-59a03e80-f57a-11e9-8e24-64940bbf2c71.png)

New rear sensor locations:
![fixed-colorback](https://user-images.githubusercontent.com/31673048/67414470-01683d00-f578-11e9-8ba7-06a822e25d4d.png)

Here's a snippet that shows how I added the spheres after calling `makeOct1axis` and `analysis1axis`
```
time = '07_02_08'
radobj.makeModule(name=name, x=module_x, y=module_y, torquetube=True, tubetype=tubetype,
	diameter=diameter, axisofrotationTorqueTube=True,
	zgap=module_zgap, ygap=module_ygap, xgap=module_xgap, numpanels=numpanels)

sceneDict1 = {'pitch':row_pitch, 'hub_height':axis_height, 'nMods':num_modules, 'nRows':num_rows}
trackerdict = radobj.makeScene1axis(trackerdict, name, sceneDict1, cumulativesky=False) 

radobj.makeOct1axis(trackerdict, time)
radobj.analysis1axis(trackerdict, time, accuracy='low', sensorsy=16)

frontx = trackerdict[time]['AnalysisObj'].x
fronty = trackerdict[time]['AnalysisObj'].y
frontz = trackerdict[time]['AnalysisObj'].z
rearx = trackerdict[time]['AnalysisObj'].rearX
reary = trackerdict[time]['AnalysisObj'].rearY
rearz = trackerdict[time]['AnalysisObj'].rearZ

for i in range(len(frontx)):
	# Add spheres at front and rear sensor locations and nudge them out from the module just a bit so they cast a larger shadow
	front_text = 'black sphere frontsensor{}'.format(i)
	front_text += '\n0\n0\n4 {} {} {} 0.008'.format(frontx[i]+0.003, fronty[i], frontz[i]+0.003)
	rear_text = 'black sphere rearsensor{}'.format(i)
	rear_text += '\n0\n0\n4 {} {} {} 0.008'.format(rearx[i]-0.003, reary[i], rearz[i]-0.003)
	frontSensorObj = radobj.makeCustomObject('FrontSensor{}'.format(i), front_text)
	radobj.appendtoScene(radfile=trackerdict[time]['radfile'], customObject=frontSensorObj, text=' ! xform -rz 0')
	rearSensorObj = radobj.makeCustomObject('RearSensor{}'.format(i), rear_text)
	radobj.appendtoScene(radfile=trackerdict[time]['radfile'], customObject=rearSensorObj, text=' ! xform -rz 0')

print("Creating OCT file for %s " % time)
radobj.makeOct1axis(trackerdict, time)
```

**Comment by costeladebarata** [comment]
@jalderman9 may ask you something? When you say modulez is hard coded elsewhere, do you mean elsewhere within your modified code or elsewhere in the original "main.py"? I added your code to my "main.py" and it complained about modulez not being defined, so I just added 0.02 instead.
PS.: I'm not a coding expert, just venturing myself here
Cheers

**Comment by jalderman9** [comment]
@costeladebarata The module thickness (what I called modulez) is set in `makeModule()` in `main.py`
What you've done should work equally well.  For my purposes I needed the ability to adjust the thickness of the module on the fly so that I could implement modules with and without frames.

**Comment by jalderman9** [comment]
I just corrected a second issue with the sensor positions.  The initial start points are incorrectly offsetn by the `xinc` and `zinc` values, but it should be 1/2 of the increment:

![sensors_incorrect](https://user-images.githubusercontent.com/31673048/68686580-af845880-0520-11ea-9f9b-5571307a0f8e.png)

![sensors_correct](https://user-images.githubusercontent.com/31673048/68686592-b4e1a300-0520-11ea-9025-37c84d4b70a7.png)

The fix is quite simple:
![diff](https://user-images.githubusercontent.com/31673048/68686890-32a5ae80-0521-11ea-98bf-e1a0db81f5ee.png)

**Comment by cdeline** [comment]
Thanks.  Fixed in #235
